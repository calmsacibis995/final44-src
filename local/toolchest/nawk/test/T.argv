echo T.argv: misc tests of argc and argv

awk=${awk-../a.out}

echo >foo1
echo >foo2
$awk '
BEGIN {
	for (i = 1; i < ARGC-1; i++)
		printf "%s ", ARGV[i]
	if (ARGC > 1)
		printf "%s", ARGV[i]
	printf "\n"
	exit
}' * >foo1
echo * >foo2
cmp foo1 foo2 || echo 'BAD: T.argv (echo1 *)'

$awk '
BEGIN {
	for (i = 1; i < ARGC; i++) {
		printf "%s", ARGV[i]
		if (i < ARGC-1)
			printf " "
	}
	printf "\n"
	exit
}' * >foo1
echo * >foo2
cmp foo1 foo2 || echo 'BAD: T.argv (echo2 *)'

$awk '
BEGIN {
	print ARGC
	ARGV[ARGC-1] = ""
	for (i=0; i < ARGC; i++)
		print ARGV[i]
	exit
}
' a bc def gh >foo1
echo "5
$awk
a
bc
def
" >foo2
diff foo1 foo2 || echo 'BAD: T.argv (argc *)'

echo '1
2
3' >foo0
echo 'foo1
foo2
foo3' >foo1
$awk '{print L $0}' L=foo <foo0 >foo2
diff foo1 foo2 || echo 'BAD: T.argv (L=foo <foo1)'

echo '1
2
3' >foo0
echo 'foo1
foo2
foo3' >foo1
$awk '{print L $0}' L=foo foo0 >foo2
diff foo1 foo2 || echo 'BAD: T.argv (L=foo foo1)'
