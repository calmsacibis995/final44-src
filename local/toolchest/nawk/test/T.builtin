echo T.builtin: test miscellaneous builtin functions

awk=${awk-../a.out}

$awk 'BEGIN { print index(123, substr(123, 2)) }' >foo1
echo 2 >foo2
diff foo1 foo2 || echo 'BAD: T.bultin (1)'
