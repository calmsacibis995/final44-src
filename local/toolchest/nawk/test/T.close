echo T.close: test close built-in

awk=${awk-../a.out}

rm -f foo
$awk '{ print >>"foo"; close "foo" }' /etc/passwd
diff /etc/passwd foo || echo 'BAD: T.close (1)'

ls -l >foo
tail -1 foo >foo1
$awk '{ print >"foo2"; close "foo2" }' foo
diff foo1 foo2 || echo 'BAD: T.close (2)'
