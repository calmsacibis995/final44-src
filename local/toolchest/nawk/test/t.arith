{ print $1, $1+$1, $1-$1, $1 * $1, $1/NR, $1 % NR, $1++, $1--, --$1, $1-- }
