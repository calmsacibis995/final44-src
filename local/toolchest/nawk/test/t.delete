{ split($0, x)
  delete x[1]
  k = 0
  for (i in x)
	k++
  if (k != NF-1)
	printf "miscount %d vs %d at %d\n", k, NF, NR
}
