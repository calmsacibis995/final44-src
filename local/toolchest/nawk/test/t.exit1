BEGIN {
	print "this is before calling abort"
	x=abort(1)
	print "this is after calling abort"
} 
func abort(n) {
	print "in abort - before exit", n
	exit 2
	print "in abort - after exit"
}
END {
	print "into END"
	x = abort(2)
	print "should not see this"
}
