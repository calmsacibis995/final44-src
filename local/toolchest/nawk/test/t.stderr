/bad/ { print | "cat 1>&2" }
	{print}
